test/http_responses_exceptions_test.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:cronet/cronet.dart';
+import 'package:test/test.dart';
+
+void main() {
+  late HttpClient client;
+  late HttpServer server;
+  setUp(() async {
+    client = HttpClient();
+    server = await HttpServer.bind(InternetAddress.anyIPv6, 5255);
+    server.listen((HttpRequest request) {
+      final paths = request.uri.pathSegments;
+      request.response.statusCode = int.parse(paths[0]);
+      request.response.close();
+    });
+  });
+
+  test('Throws a HttpException: Not Found when status code is 404', () async {
```

Comment:
Having group can be beneficial here, it reduces the test description you would have to write.
```dart
group(' HttpException', (){
  test('404, Not Found',()async{
    ...
  })
});
```